core/src/main/java/io/questdb/griffin/engine/table/ShowAllTablePartitionsCursoryFactory.java

```diff
@@ -0,0 +1,321 @@
+    private static final int ROW_COUNT;
+    private static final int SIZE_B;
+    private static final RecordMetadata METADATA;
+    private static final Comparator<String> CHAR_COMPARATOR = Chars::compare;
+    private final AllTablePartitionListRecordCursor cursor =
+            new AllTablePartitionListRecordCursor();
+    private static final Log LOG = LogFactory.getLog(AllTablePartitionListRecordCursor.class);
+    private CairoConfiguration cairoConfig;
+    private FilesFacade ff;
+
+
+    public ShowAllTablePartitionsCursoryFactory() {
+        super(METADATA);
+    }
+
+    @Override
+    public RecordCursor getCursor(SqlExecutionContext executionContext) throws SqlException {
+        cursor.of(executionContext);
+        this.cairoConfig = executionContext.getCairoEngine().getConfiguration();
+        this.ff = cairoConfig.getFilesFacade();
+        return cursor.initialize();
+    }
+
+    @Override
+    public boolean recordCursorSupportsRandomAccess() {
+        return false;
+    }
+
+    private class AllTablePartitionListRecordCursor implements NoRandomAccessRecordCursor {
+        private final ObjHashSet<TableToken> tableBucket = new ObjHashSet<>();
+        ObjList<String> attachablePartitions = new ObjList<>(4);
+        ObjList<String> detachedPartitions = new ObjList<>(8);
+        Path path = new Path();
+        private int tableIndex = -1;
+        StringSink partitionName = new StringSink();
+        private TxReader detachedTxReader;
+        private TableReaderMetadata detachedMetaReader;
+        private AllPartitionsRecord record = new AllPartitionsRecord();
+        private SqlExecutionContext executionContext;
+        private TableReader tableReader;
+        private int rootLen;
+
+        private AllTablePartitionListRecordCursor initialize() {
+            return this;
+        }
+
+        @Override
+        public void close() {
+        }
+
+        private void of(SqlExecutionContext executionContext) {
+            this.executionContext = executionContext;
+            toTop();
+        }
+
+        @Override
+        public Record getRecord() {
+            return record;
+        }
+
+        @Override
+        public boolean hasNext() throws DataUnavailableException {
+            if (tableIndex < 0) {
+                executionContext.getCairoEngine().getTableTokens(tableBucket, false);
+                tableIndex = -1;
+            }
+            tableIndex++;
+            int n = tableBucket.size();
+            for (; tableIndex < n; tableIndex++) {
+                TableToken tableToken = tableBucket.get(tableIndex);
+                if (tableToken.isSystem())
+                    continue;
+                record.loadPartitionDetailsForCurrentTable(tableToken);
+                return true;
+            }
+            return tableIndex < n;
+
+        }
+
+        @Override
+        public long size() throws DataUnavailableException {
+            return 0;
```

Comment:
see parent interface comments: Not every record cursor has a size, may return -1.
If we return 1 row per table, we know the size 
```suggestion
            return tableBucket.size();
```